### DISCUSSION TITLE
CodeIgniter\Security\Security::generateHash()

### DISCUSSION THREAD

**Discussion by geriweb** [codeigniter4/shield#749] [Q&A]
The CodeIgniter\Security\Security::generateHash() function should be public, because it is now drop an error message, with session authentication:
Call to protected method CodeIgniter\Security\Security::generateHash() from scope CodeIgniter\Shield\Authentication\Authenticators\Session

**Comment by kenjis** [comment]
It is public.
See https://codeigniter4.github.io/CodeIgniter4/changelogs/v4.2.3.html?highlight=generatehash

And Shield requires CodeIgniter 4.2.7+.
See https://codeigniter4.github.io/shield/install/#requirements

**Reply by geriweb** [reply]
OK, I have to update CodeIgniter, thanks!
